source/world/level/TilePos.cpp

```diff
@@ -0,0 +1,132 @@
+#include "TilePos.hpp"
+#include "world/phys/Vec3.hpp"
+#include "world/level/levelgen/chunk/ChunkPos.hpp"
+
+TilePos::TilePos()
+	: TilePos(0, 0, 0) {}
+
+TilePos::TilePos(int _x, int _y, int _z)
+	: x(_x), y(_y < 0 ? 0 : _y), z(_z) {}
+
+TilePos::TilePos(float _x, float _y, float _z)
+	: TilePos((int)floorf(_x), (uint8_t)floorf(_y), (int)floorf(_z)) {}
```

Comment:
how about using `Mth::floor` here

Reply:
Because Mojang used floorf here. Is there any *good* reason for me to be using Mth::floor over floorf?